docs/next/markdoc/nodes/heading.markdoc.ts

```diff
@@ -0,0 +1,32 @@
+import {Tag} from '@markdoc/markdoc';
+
+import {Heading} from '../../components/Heading';
+
+function generateID(children, attributes) {
+  if (attributes.id && typeof attributes.id === 'string') {
+    return attributes.id;
+  }
+  return children
+    .filter((child) => typeof child === 'string')
+    .join(' ')
+    .replace(/[?]/g, '')
+    .replace(/\s+/g, '-')
+    .toLowerCase();
```

Comment:
Maybe worth using `slugify`? https://www.npmjs.com/package/slugify (Unless this is what we're already doing for slug generation and making a change would break search results or something unfortunate like that.)

Reply:
this was directly copied over from markdoc's example. unfortunately, slugify doesnt provide esm version.

but i looked in how we currently do it: we currently use https://github.com/rehypejs/rehype-slug which is a plugin of rehype, and to keep our new renderer simple, i'd like to remove that from our dependency. so to keep the same slug generation logic (rehype-slug implementation [here](https://github.com/rehypejs/rehype-slug/blob/main/lib/index.js)), we could maybe use the package it uses: https://github.com/Flet/github-slugger